lib/connections/service.go

```diff
@@ -370,15 +382,16 @@ func (s *Service) connect() {
 	}
 }
 
-func (s *Service) shouldLimit(addr net.Addr) bool {
-	if s.cfg.Options().LimitBandwidthInLan {
+func (s *Service) shouldLimit(addr net.Addr, options config.OptionsConfiguration) bool {
+	if options.LimitBandwidthInLan {
```

Comment:
So to me this whole thing is bleeding functionality where it's not supposed to be.
We explicitly pass in a config object just for the purpose of checking a single boolean on the first line and then immediately returning.

Instead we could not pass anything, save a stack frame (both real and mental), and potentially only have to compare one part of an if expression.

Maybe that's fine in Java school, but in average joe code where we don't have a stupid number of abstractions this just feels wrong.